fdbclient/NativeAPI.actor.cpp

```diff
@@ -5862,8 +5871,13 @@ ACTOR void checkWrites(Reference<TransactionState> trState,
 ACTOR static Future<Void> commitDummyTransaction(Reference<TransactionState> trState, KeyRange range) {
 	state Transaction tr(trState->cx);
 	state int retries = 0;
-	state Span span("NAPI:dummyTransaction"_loc, trState->spanID);
-	tr.span.addParent(span.context);
+	// TODO - ljoswiak, mpilman. This will require some inspection. Here we're setting the parent context.
+	// Before you used addParent, however there is a side effect in that old function, due to no links.
+	// You would check if there is no parent set and if so, you'd overwrite the traceids.
+	//
+	// Need to determine if this behavior matches previously intended, here we'd just always set parent.
+	state Span span("NAPI:dummyTransaction"_loc, trState->spanContext);
+	tr.span.addParentOrLink(span.context);
```

Comment:
I think simplifying this to just add a parent in all cases should be ok. Since the intention seems to be add a parent to the newly created transaction which should have no parents at this point, I guess we can assume there is nothing to overwrite in the child at this point. So `addParentOrLink` can probably be simplified to just `addParent`. It's possible we might have to revisit some of the span logic in the future though.

Reply:
OK, let's talk about this in review today, makes sense to me though fwiw.